lisa/sut_orchestrator/libvirt/ch_platform.py

```diff
@@ -57,13 +62,25 @@ def _configure_node(
 
         assert isinstance(node_runbook, CloudHypervisorNodeSchema)
         node_context = get_node_context(node)
-        if self.host_node.is_remote:
-            node_context.firmware_source_path = node_runbook.firmware
-            node_context.firmware_path = os.path.join(
-                self.vm_disks_dir, os.path.basename(node_runbook.firmware)
-            )
+        if node_runbook.kernel:
+            if self.host_node.is_remote:
+                if node_runbook.kernel.is_remote_path:
+                    node_context.kernel_path = node_runbook.kernel.path
+                else:
+                    node_context.kernel_source_path = node_runbook.kernel.path
+                    node_context.kernel_path = os.path.join(
+                        self.vm_disks_dir, os.path.basename(node_runbook.kernel.path)
+                    )
+            else:
+                node_context.kernel_path = node_runbook.kernel.path
```

Comment:
```suggestion
            if self.host_node.is_remote and not node_runbook.kernel.is_remote_path:
                node_context.kernel_source_path = node_runbook.kernel.path
                node_context.kernel_path = os.path.join(
                    self.vm_disks_dir, os.path.basename(node_runbook.kernel.path)
                )
            else:
                node_context.kernel_path = node_runbook.kernel.path
```

Reply:
Thanks for the suggestion, I'll make the change and push the update shortly.